### DISCUSSION TITLE
Enable new push notifications functionality

### DISCUSSION THREAD

**Discussion by Dakael** [hassio-addons/addon-bitwarden#279] [Ideas]
# Problem/Motivation

Now that vaultwarden supports [push notifications](https://github.com/dani-garcia/vaultwarden/pull/3304) for syncing mobile bitwarden clients, we need to be able to pass the following environment variables to the addon container enable it.

```
PUSH_ENABLED=true
PUSH_INSTALLATION_ID=CHANGEME
PUSH_INSTALLATION_KEY=CHANGEME
```
Unfortunately this cannot be done through the admin page, as the three relevant fields are in the Read Only Config section.

![image](https://github.com/hassio-addons/addon-bitwarden/assets/21049524/46def99e-f077-442e-b745-41fdf9aea88c)

## Proposed changes

Ideally, add the three as options in the addon configuration screen in home assistant.

**Comment by frenck** [comment]
> Ideally, add the three as options in the addon configuration screen in home assistant.

That is a hard "no" for me. I suggest opening up a feature request upstream to get it added to the admin page. As all other configuration is there, putting these there makes much more sense as well.

../Frenck

**Reply by xescure** [reply]
Since it's not happening upstream are you open to a PR?

**Comment by frenck** [comment]
Converted this issue into a discussion, as it is not a bug.

../Frenck

**Comment by phobiac** [comment]
Not sure if this is the appropriate space for this or not, but I'm posting here since it's related. Ignoring that the push notification doesn't work when I try to manually approve the pending login request I get a "We were unable to process your request." message from the Android client and this in the addon logs. I've removed the actual key and deviceIdentifier values, but I'm assuming the masterPasswordHash being null is tripping this up. I'm not sure if this is something wrong with my setup though.

`2023-09-15 00:57:04.285][vaultwarden::api::core::accounts::_][WARN] Data guard `Json < AuthResponseRequest >` failed: Parse("{\"key\":\"****",\"masterPasswordHash\":null,\"deviceIdentifier\":\"****",\"requestApproved\":true}", Error("invalid type: null, expected a string", line: 1, column: 381)).
[2023-09-15 00:57:04.285][response][INFO] (put_auth_request) PUT /api/auth-requests/<uuid> => 422 Unprocessable Entity`

**Comment by phobiac** [comment]
There seems to be no interest on the vaultwarden end in making this configurable through the admin page

https://github.com/dani-garcia/vaultwarden/discussions/2469#discussioncomment-7095834

**Comment by minmaxat** [comment]
i opend a new discussion for this topic in Vaulwarden: 

https://github.com/dani-garcia/vaultwarden/discussions/4180

**Reply by Getslow6** [reply]
There might be a workaround, if someone can try on a developer instance of the Home Assistant addon:

https://github.com/dani-garcia/vaultwarden/discussions/4180#discussioncomment-8232502
